roadmap.js

```diff
@@ -0,0 +1,354 @@
+            let support = features[featName];
+            let box, note;
+
+            // First extract the footnote part if it's an array
+            if (Array.isArray(support)) {
+              note = support[1];
+              support = support[0];
+            }
+
+            if (typeof support === 'string') {
+              if (support === 'flag') {
+                // 'flag' is treated specially as the "requires flag" icon
+                box = buildCellInner('flag');
+              } else {
+                // Otherwise it's a version number, like "95"
+                box = buildCellInner('yes', support);
+                if (!note) note = `✓ Supported since version ${support}`;
+              }
+            } else if (!support) {
+              if (support === null) {
+                box = buildCellInner('na', 'N/A');
+                if (!note) note = '✗ Not applicable for this browser/engine';
+              } else {
+                box = buildCellInner('no');
+                if (!note) note = '✗ Not supported';
+              }
+            } else {
+              if (support !== true) throw new TypeError();
+              box = buildCellInner('yes');
+              // Magic value, keep in sync with `renderNote`
+              if (!note) note = '✓ Supported, introduced in unknown version';
+            }
+
+            const cell =  h('td', {
+              headers: [idMap['table-col'](browserName), idMap['table-row'](featName)].join(' ')
+            }, [box]);
+
+            if (note && note2index.has(note)) {
+              cell.tabIndex = 0;  // focusable
+              const index = note2index.get(note);
+              const [noteId, refLink] = createNoteRef(index);
+              box.appendChild(h('sup', {}, [refLink]));
+
+              const noteItem = document.getElementById(noteId);
+              if (noteItem) {
+                cell.addEventListener('mouseenter', () => noteItem.classList.add('ref-highlight'));
+                cell.addEventListener('mouseleave', () => noteItem.classList.remove('ref-highlight'));
+              }
+            }
+
+            // Clip to both <tbody> and the scrollbox.
+            // the former is to avoid blocking out the headers;
+            // the latter is to keep the tooltip inside the scrollable area
+            addTooltip(cell, note, [tBody, scrollbox]);
+            return cell;
+          })
+        ])
+      );
+      tBody.lastElementChild.setAttribute('aria-describedby', idMap['table-row'](featName));
+    }
+  }
+
+  function buildCellInner(type, text) {
+    const content = text || icon(type);
+    return h('div', { className: `feature-cell icon-${type}`}, [content]);
+  }
+
+  function renderNote(note) {
+    // No point for screen readers to pronounce those symbols out loud.
+    for (const symbol of ['✓', '✗']) {
+      if (note.startsWith(symbol)) {
+        note = `<span aria-hidden='true'>${symbol}</span>${note.substring(1)}`;
```

Comment:
Nit: why does this one use raw HTML via `.innerHTML`, unlike the rest that use `h` API?

Reply:
Ah, if it's because of text transform below, perhaps it can be moved around? I'd really rather avoid worrying about escaping HTML-unsafe chars :)